test/server/graphql/v2/mutation/UploadedFileMutations.test.ts

```diff
@@ -0,0 +1,148 @@
+          const result = await graphqlQueryV2(uploadFileMutation, args, user);
+          expect(result.errors).to.exist;
+          expect(result.errors[0].message).to.eq(
+            `parseDocument is only supported for EXPENSE_ITEM and EXPENSE_ATTACHED_FILE`,
+          );
+        }
+      });
+
+      it('does not crash but returns a proper error if parsing fails', async () => {
+        // Mock OCR service with something that always fails
+        sandbox.stub(ExpenseOCRLib, 'getExpenseOCRParser').returns({
+          processUrl: () => {
+            throw new Error('OCR parsing failed on purpose for test');
+          },
+          processImage: () => {
+            throw new Error('OCR parsing failed on purpose for test');
+          },
+        });
+
+        const user = await fakeUser();
+        const args = { kind: 'EXPENSE_ITEM', file: getMockFileUpload(), parseDocument: true };
+        const result = await graphqlQueryV2(uploadFileMutation, args, user);
+        expect(result.errors).to.not.exist;
+        expect(result.data.uploadFile.parsingResult).to.deep.eq({
+          success: false,
+          message: 'Could not parse document: OCR parsing failed on purpose for test',
+          confidence: null,
+          expense: null,
+        });
+      });
+
+      // Since Klippa is not configured, the mock service is used by default
+      describe('mocked service', () => {
+        it('is used by default', async () => {
+          const user = await fakeUser();
+          const args = { kind: 'EXPENSE_ITEM', file: getMockFileUpload(), parseDocument: true };
+          const result = await graphqlQueryV2(uploadFileMutation, args, user);
+          expect(result.errors).to.not.exist;
+          expect(result.data.uploadFile.parsingResult).to.not.be.null;
+          expect(result.data.uploadFile.parsingResult).to.deep.eq({
+            success: true,
+            message: null,
+            confidence: 100,
+            expense: {
+              description: 'Mock description',
+              amount: { valueInCents: 100e2, currency: 'USD' },
+              incurredAt: '2020-02-01',
+            },
+          });
+        });
+
+        it('is not used in production', async () => {
+          sandbox.stub(config, 'env').value('production');
+          const user = await fakeUser();
+          const args = { kind: 'EXPENSE_ITEM', file: getMockFileUpload(), parseDocument: true };
+          const result = await graphqlQueryV2(uploadFileMutation, args, user);
+          expect(result.errors).to.not.exist;
+          expect(result.data.uploadFile.parsingResult).to.deep.eq({
+            success: false,
+            message: 'OCR parsing is not available',
+            confidence: null,
+            expense: null,
+          });
+        });
+      });
+
+      // TODO Mock Klippa service and make sure we call its endpoints here
+      // describe('using Klippa', () => {
```

Comment:
When you implement this, there's a helper to write integration tests using nock `useIntegrationTestRecorder`.

Reply:
Interesting!